python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -164,74 +172,100 @@ def run(self, templateExposure, exposure, subtractedExposure, psfMatchingKernel)
+        self.log.info("Variance (science): %f" % var1)
+        if var2 is None:
+            statObj = afwMath.makeStatistics(templateExposure.getMaskedImage().getVariance(),
+                                             templateExposure.getMaskedImage().getMask(),
+                                             afwMath.MEANCLIP, self.statsControl)
+            var2 = statObj.getValue(afwMath.MEANCLIP)
+        self.log.info("Variance (template): %f" % var2)
+
+        statObj = afwMath.makeStatistics(subtractedExposure.getMaskedImage().getVariance(),
+                                         subtractedExposure.getMaskedImage().getMask(),
+                                         afwMath.MEANCLIP, self.statsControl)
+        var = statObj.getValue(afwMath.MEANCLIP)
+        self.log.info("Variance (uncorrected diffim): %f" % var)
+
+        corrKernel = DecorrelateALKernelTask._computeDecorrelationKernel(kimg.getArray(), var1, var2)
         self.log.info("Convolving.")
-        correctedExposure, corrKern = DecorrelateALKernelTask._doConvolve(subtractedExposure, fcorrKernel)
+        correctedExposure, corrKern = DecorrelateALKernelTask._doConvolve(subtractedExposure, corrKernel)
         self.log.info("Updating correctedExposure and its PSF.")
 
         # Compute the subtracted exposure's updated psf
         psf = subtractedExposure.getPsf().computeImage().getArray()
-        psfc = DecorrelateALKernelTask.computeCorrectedDiffimPsf(fcorrKernel, psf, sig1=sig1, sig2=sig2)
+        psfc = DecorrelateALKernelTask.computeCorrectedDiffimPsf(corrKernel, psf, var1=var1, var2=var2)
         psfcI = afwImage.ImageD(psfc.shape[0], psfc.shape[1])
         psfcI.getArray()[:, :] = psfc
         psfcK = afwMath.FixedKernel(psfcI)
         psfNew = measAlg.KernelPsf(psfcK)
         correctedExposure.setPsf(psfNew)
         self.log.info("Complete.")
 
+        statObj = afwMath.makeStatistics(correctedExposure.getMaskedImage().getVariance(),
+                                         correctedExposure.getMaskedImage().getMask(),
+                                         afwMath.MEANCLIP, self.statsControl)
+        var = statObj.getValue(afwMath.MEANCLIP)
+        self.log.info("Variance (corrected diffim): %f" % var)
+
         return pipeBase.Struct(correctedExposure=correctedExposure, correctionKernel=corrKern)
 
     @staticmethod
-    def _computeDecorrelationKernel(kappa, sig1=0.2, sig2=0.2):
+    def _computeDecorrelationKernel(kappa, var1=0.04, var2=0.04):
         """! Compute the Lupton/ZOGY post-conv. kernel for decorrelating an
         image difference, based on the PSF-matching kernel.
         @param kappa  A matching kernel 2-d numpy.array derived from Alard & Lupton PSF matching
-        @param sig1   Average sqrt(variance) of template image used for PSF matching
-        @param sig2   Average sqrt(variance) of science image used for PSF matching
+        @param var1   Average variance of science image used for PSF matching
```

Comment:
Same thing here: if var1==science, then call it science. Otherwise don't call anything science or template and just use 1 and 2 everywhere.